app/bundles/DynamicContentBundle/Helper/DynamicContentHelper.php

```diff
@@ -118,4 +137,110 @@ public function replaceTokensInContent($content, $lead)
 
         return $content;
     }
+
+    /**
+     * @param string    $content
+     * @param Lead|null $lead
+     *
+     * @return array
+     */
+    public function findDwcTokens($content, $lead)
+    {
+        preg_match_all('/{dwc=(.*?)}/', $content, $matches);
+
+        $tokens = [];
+        if (!empty($matches[1])) {
+            foreach ($matches[1] as $key => $slotName) {
+                $token = $matches[0][$key];
+                if (!empty($tokens[$token])) {
+                    continue;
+                }
+
+                $dwcs = $this->getDwcsBySlotName($slotName);
+
+                /** @var DynamicContent $dwc */
+                foreach ($dwcs as $dwc) {
+                    if ($dwc->getIsCampaignBased()) {
+                        continue;
+                    }
+                    $content                   = $lead ? $this->getRealDynamicContent($dwc->getSlotName(), $lead, $dwc) : '';
+                    $tokens[$token]['content'] = $content;
+                    $tokens[$token]['filters'] = $dwc->getFilters();
```

Comment:
It is better to use structured object instead of array:
```
tokens = new Token(); // for line 151
tokens->add($content, $dwc->getFilters());

/***/
class Token {
private $items;

public function add($content, xx $filter) {
$this->items[] = [...];
}
..
}
```
This is the easiest implementation, even better is to have another class for content of Token: https://github.com/mautic/mautic/pull/5082/commits/8287eb543bdb79d1c693dac024d72672d5d77201#diff-f18858aaed88e15db191fa7ffa299f45

I had this simple data class and change it to another structure in this commit because I need to hold more data.